pkg/tools/etcd_helper_watch.go

```diff
@@ -250,6 +259,10 @@ func (w *etcdWatcher) translate() {
 			return
 		case res, ok := <-w.etcdIncoming:
 			if ok {
+				if curLen := len(w.etcdIncoming); curLen > 0 {
```

Comment:
Basically I want to know if it's ever > 0-- if so, that justifies this patch. If not, this patch didn't fix anything.

I can adjust to reduce log spam, but the decoding process that is backing things up is *much* more resource intensive than a log message, so this is "only" a cosmetic problem.